### PR TITLE
[vrotsc,vrotsc-annotations] (#324) Add Workflow canvas Error Handling

### PR DESCRIPTION
**Pull request by Michaelpalacce**
### Description

TBD

### Checklist

- [x] I have added relevant error handling and logging messages to help troubleshooting
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] I have added necessary documentation, relevant usage information (if applicable)
- [x] I have updated the PR title with affected component, related issue number and a short summary of the changes introduced
- [x] I have added labels for implementation kind (kind/*) and version type (version/*)
- [x] I have tested against live environment, if applicable
- [x] I have synced any structure and/or content vRA-NG improvements with vra-ng and ts-vra-ng archetypes (if applicable)
- [x] I have my changes rebased and squashed to the minimal number of relevant commits. **Notice: don't squash all commits**
- [x] I have added a descriptive commit message with a short title, including a `Fixed #XXX -` or `Closed #XXX -` prefix to auto-close the issue

### Testing

e2e test added

### CODE REVIEW

**Comment by akantchev** [comment]
Hi @Michaelpalacce,

I did some tests with a real environment, however when I try to build a package an push it to an environment the following error occurs:

```
[ERROR] /home/alex/gitlab/Projects/vro.ts.workflows/node_modules/@vmware-pscoe/vropkg/node_modules/sax/lib/sax.js:651
[ERROR]     er = new Error(er)
[ERROR]          ^
[ERROR]
[ERROR] Error: No whitespace between attributes
[ERROR] Line: 14
[ERROR] Column: 555
[ERROR] Char: b
[ERROR]     at error (/home/alex/gitlab/Projects/vro.ts.workflows/node_modules/@vmware-pscoe/vropkg/node_modules/sax/lib/sax.js:651:10)
[ERROR]     at strictFail (/home/alex/gitlab/Projects/vro.ts.workflows/node_modules/@vmware-pscoe/vropkg/node_modules/sax/lib/sax.js:677:7)
[ERROR]     at SAXParser.write (/home/alex/gitlab/Projects/vro.ts.workflows/node_modules/@vmware-pscoe/vropkg/node_modules/sax/lib/sax.js:1395:13)
[ERROR]     at new XmlDocument (/home/alex/gitlab/Projects/vro.ts.workflows/node_modules/@vmware-pscoe/vropkg/node_modules/xmldoc/lib/xmldoc.js:301:19)
[ERROR]     at exports.xml (/home/alex/gitlab/Projects/vro.ts.workflows/node_modules/@vmware-pscoe/vropkg/dist/parse/util.js:14:25)
[ERROR]     at Object.getWorkflowItems (/home/alex/gitlab/Projects/vro.ts.workflows/node_modules/@vmware-pscoe/vropkg/dist/parse/util.js:279:62)
[ERROR]     at parseInputForms (/home/alex/gitlab/Projects/vro.ts.workflows/node_modules/@vmware-pscoe/vropkg/dist/parse/tree.js:88:28)
[ERROR]     at parseTreeElement (/home/alex/gitlab/Projects/vro.ts.workflows/node_modules/@vmware-pscoe/vropkg/dist/parse/tree.js:64:24)
[ERROR]     at /home/alex/gitlab/Projects/vro.ts.workflows/node_modules/@vmware-pscoe/vropkg/dist/parse/tree.js:112:26
[ERROR]     at Array.map (<anonymous>)
[ERROR]     at Object.<anonymous> (/home/alex/gitlab/Projects/vro.ts.workflows/node_modules/@vmware-pscoe/vropkg/dist/parse/tree.js:112:14)
[ERROR]     at Generator.next (<anonymous>)
[ERROR]     at /home/alex/gitlab/Projects/vro.ts.workflows/node_modules/@vmware-pscoe/vropkg/dist/parse/tree.js:8:71
[ERROR]     at new Promise (<anonymous>)
[ERROR]     at __awaiter (/home/alex/gitlab/Projects/vro.ts.workflows/node_modules/@vmware-pscoe/vropkg/dist/parse/tree.js:4:12)
[ERROR]     at Object.parseTree (/home/alex/gitlab/Projects/vro.ts.workflows/node_modules/@vmware-pscoe/vropkg/dist/parse/tree.js:109:12)
[ERROR]
[ERROR] Node.js v22.4.1
[INFO] Running vropkg... finished
[INFO] ------------------------------------------------------------------------
[INFO] BUILD FAILURE
[INFO] ------------------------------------------------------------------------
```
It appears that somewhere when you generate the XML there is a missing white-space, hence the XML is incorrect, could you please check.

I used the following workflow code:
```ts
import { Workflow, Out, In, Err, Item, RootItem, DecisionItem, WaitingTimerItem, WorkflowEndItem, DefaultErrorHandler } from "vrotsc-annotations";

@Workflow({
    name: "Complex Workflow",
    path: "VMware/PSCoE",
    attributes: {
        waitingTimer: {
            type: "Date"
        },
        counter: {
            type: "number"
        }
    }
})
export class Complex {

    @DecisionItem({
        target: "waitForEvent",
        else: "end"
    })
    public decisionElement(waitingTimer: Date) {
        return waitingTimer !== null;
    }

    @Item({
        target: "decisionElement",
        exception: ""
    })
    public execute(
        @Out @In waitingTimer: Date,
        @Out @In counter: number
    ): void {
        if (!counter) {
            counter = 0;
        }
        counter++;
        if (counter < 2) {
            const tt = Date.now() + 5 * 1000;
            waitingTimer = new Date(tt);
        } else {
            waitingTimer = null;
        }

        System.log("Counter: " + counter);
        System.log("Waiting Timer: " + waitingTimer);
    }

    @Item({
        target: "execute",
        exception: ""
    })
    @RootItem()
    public start() {
        System.log("Starting workflow");
    }

    @WaitingTimerItem({
        target: "execute"
    })
    public waitForEvent(@In waitingTimer: Date) {
        // NOOP
    }

	@DefaultErrorHandler({
		target: "workflowEnd"
	})
	public defaultErrorHandler(@Out errorMessage: string) {
		// NOOP
	}

    @WorkflowEndItem({
        endMode: 1,
        businessStatus: "Bad"
    })
    public workflowEnd(@Err errorMessage: string) {
        // NOOP
    }
}
```

File: docs/versions/latest/Release.md [hunk 1]
```diff
@@ -24,6 +24,55 @@
 [//]: # (Optional But higlhy recommended Specify *NONE* if missing)
 [//]: # (#### Relevant Documentation:)
 
+### *Error handling for workflow canvas elements*
+
+There is a new decorator added `@Err` that can be used with arguments. This decorator will mean that the argument will be bound to the exeption if it occurs.
+Note, you add is as normal `@In` or `@Out` decorators, but the variable should not be used in the method.
```
Comment (VenelinBakalov) [inline docs/versions/latest/Release.md:30]:
```suggestion
Note, you add it as normal `@In` or `@Out` decorators, but the variable should not be used in the method.
```

**Comment by VenelinBakalov** [comment]
> @akantchev I think you need to define: `errorMessage` as an attribute too:
> 
> attributes: { waitingTimer: { type: "Date" }, counter: { type: "number" } }

@akantchev please check whenever you can and if everything is ok, let's resolve the previous comment

**Review by VenelinBakalov — CHANGES_REQUESTED** [review]
Just merge conflict resolution left

**Review by VenelinBakalov — APPROVED** [review]

**Pull request merged**